lib/ecto/association.ex

```diff
@@ -627,6 +627,12 @@ defmodule Ecto.Association.Has do
       raise ArgumentError, "expected `:where` for #{inspect name} to be a keyword list, got: `#{inspect where}`"
     end
 
+    preload_order = opts[:preload_order] || []
+
+    unless is_list(preload_order) do
```

Comment:
The format of the `preload_order` should be validated?

Reply:
Good question … of course it's up to maintainers, but I guess that if we have `where` validation looks same then it's all ok.

Reply:
Good question … of course it's up to maintainers, but I guess that if `where` validation looks same then it's all ok.

Reply:
It should. It should be a list of atoms and 2-elem tuples where the first element is either desc or asc.

Reply:
If you can validate the field exists, then it is even better!

Reply:
> element is either desc or asc

@josevalim The other directions should also be supported,right? Like `asc_nulls_first` and `desc_nulls_last`.

> If you can validate the field exists, then it is even better!

Any direction on how we could implement this? My first instinct was to call `related.__schema__(:fields)`, but the `__schema__` seems to be available only at runtime.

Reply:
> @josevalim The other directions should also be supported,right? Like asc_nulls_first and desc_nulls_last.

Yes, good call!

> Any direction on how we could implement this? My first instinct was to call related.__schema__(:fields), but the __schema__ seems to be available only at runtime.

Hrm, I misremembered. We actually don't validate fields yet. So don't worry about it, just validate the format. :)

Reply:
Maybe we can do it in the __after_compile__ calback? Not sure about it, though.

Reply:
Maybe we can do it in the after_compile calback? Not sure about it, though.

Reply:
@josevalim Do you think we should add tests for `asc_nulls_first` and `desc_nulls_first` as well?